__tests__/types/typesCheck.js

```diff
@@ -0,0 +1,54 @@
+const ts = require('typescript');
+const path = require('path');
+
+function compile(...filenames) {
+const program = ts.createProgram(filenames, {
+  allowSyntheticDefaultImports: true,
+  jsx: 'react',
+  noEmit: false,
+  skipLibCheck: true,
+});
+
+const emitResult = program.emit();
+const allDiagnostics = ts.getPreEmitDiagnostics(program).concat(emitResult.diagnostics);
+const errors = [];
+
+allDiagnostics.forEach(diagnostic => {
+  if (diagnostic.file) {
+    const { line, character } = diagnostic.file.getLineAndCharacterOfPosition(diagnostic.start);
+    const message = ts.flattenDiagnosticMessageText(diagnostic.messageText, '\n');
+    errors.push(`${diagnostic.file.fileName} (${line + 1},${character + 1}): ${message}`);
+  } else {
+    errors.push(ts.flattenDiagnosticMessageText(diagnostic.messageText, '\n'));
+  }
+});
+
+if (errors.length > 0)
+
+return errors;
+};
+
+
+it('dir should pass dir as string', () => {
```

Comment:
```suggestion
it('should pass dir as string', () => {
```

So it can read as a sentence "it should pass dir as string". This is called BDD.